### PR TITLE
Add additional documentation for Python, Rego, and YAML contribs

### LINKED ISSUE #54: Add more detail around contrib scenarios
**What would you like to be added**:

Need to add additional documentation around different contributor scenarios.

Examples:
- [X] Run linting/formatting if you edit Python files (#80)
- [X] Run linting/formatting if you edit Rego files (#60/#80)
- [X] Rebuild install manifest if you edit YAML manifests (#80)

**Why is this needed**:

Lower barrier to entry for new contributors and to help me not have to remember!

### PR DESCRIPTION
**Pull request by phenixblue**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design

/kind documentation
> /kind feature

**What this PR does / why we need it**:

Since we've matured several different processes related to CI and built out associated Make targets to provide macros for common contributor tasks, we needed to update our Contributor docs.

These updates include specific tasks such as code formatting, linting, and unit tests for Python, Rego, and BASH.

**Which issue(s) this PR fixes**:

Fixes #54

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., usage docs, etc.**:

```docs

```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,13 +22,51 @@ We don't follow a strict coding style, but we do have a few preferences:
...
+```shell
+# Lint and apply formatting
+$ make lint-shell
+```
+
+## YAML Manifests
+
+The YAML manifests to manage the installation and configuration of MagTape are stored in the [/deploy/manifests](/deploy/manifests) directory. These are managed separately for readability and maximum flexibility in configuration (Kustomize), but they are also combined together with some sane defaults into one single install manifest at [/deploy/install.yaml](/deploy/install.yaml) for an easy on-liner POC install.
+
+There are some Kustomize files related to the advanced install process documented [here](/docs/install.md). If you make changed to any of the manifests, please maintain functionality with Kustomize in a way that is accurate the docs (or update the docs where applicable).
```
Comment (ilrudie) [inline CONTRIBUTING.md:61]: Seems like you are missing a word. Should this be "...in a way that is accurate **_to_** the docs..." instead?

Comment (phenixblue) [reply CONTRIBUTING.md:61]: Yeah, good catch. I also found another typo on a closer read through:

```
If you make changed to any of the
```

`changed` should be `changes`

Will get a fix pushed soon

Comment (ilrudie) [reply CONTRIBUTING.md:61]: lol, how many software guys does it take to proofread a markdown doc?

Comment (phenixblue) [reply CONTRIBUTING.md]: Ok, these spelling mistakes are fixed.

**Review by ilrudie — APPROVED** [review]
lgtm

**Pull request merged**